pilot/pkg/model/virtualservice.go

```diff
@@ -429,6 +429,35 @@ func mergeHTTPMatchRequest(root, delegate *networking.HTTPMatchRequest) *network
 	return out
 }
 
+func cloneHTTPMatchRequest(delegate *networking.HTTPMatchRequest) *networking.HTTPMatchRequest {
+	copied := &networking.HTTPMatchRequest{}
```

Comment:
should we use `proto.Clone`? Might be more reliable?

Reply:
I started with that first but later changed to this thinking that this performs better.

Reply:
like https://github.com/istio/istio/pull/47214/files#diff-265bcfd360a47f223b24a3151abaed4b59145e8b1923dd3e68e0518bd2e50dbeR384

Here we only need shallow copy

Reply:
I do not know if it is safe with protobuf with locks etc?